src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,606 @@
+    dict
+        Various computed information about the video.
+    """
+
+    input_h5 = data.get('input_h5', None)
+    if input_h5 is None:
+        raise KeyError("input JSON does not have required field: 'input_h5'")
+    if not os.path.exists(input_h5):
+        raise IOError(f"{input_h5} does not exist")
+
+    output_h5 = data.get('output_h5', None)
+    if output_h5 is None:
+        raise KeyError("input JSON does not have required field: 'output_h5'")
+
+    equipment_name = data.get("equipment_name", None)
+    if equipment_name is None:
+        raise KeyError("input JSON does not have"
+                       "required field: 'equipment_name'")
+
+    aLstr = data.get("aL", None)
+    if aLstr is None:
+        raise KeyError(f"parameter aL not determined for {equipment_name}")
+    aL = float(aLstr)
+
+    aRstr = data.get("aR", None)
+    if aRstr is None:
+        raise KeyError(f"parameter aR not determined for {equipment_name}")
+    aR = float(aRstr)
+
+    bLstr = data.get("bL", None)
+    if bLstr is None:
+        raise KeyError(f"parameter bL not determined for {equipment_name}")
+    bL = float(bLstr)
+
+    bRstr = data.get("bR", None)
+    if bRstr is None:
+        raise KeyError(f"parameter bR not determined for {equipment_name}")
+    bR = float(bRstr)
+
+    logging.debug(f"Equipment name: {str(equipment_name)}")
+    logging.debug(f"aL: {aL}   aR: {aR}")
+    logging.debug(f"bL: {bL}   bR: {bR}")
+
+    return input_h5, output_h5, aL, aR, bL, bR
+
+
+def make_output_file(output_file: str,
+                     output_dataset: str,
+                     xtable: dict,
+                     FOVwidth: int,
+                     movie_shape: (int, int),
+                     movie_dtype):
+    """
+    Creates the output h5 file, and ensures that it has a Dataset
+    of the correct size for the dewarped movie.
+
+    Parameters
+    ----------
+    output_file: str
+        File path where the h5 containing the dewarped video will be saved.
+    output_dataset: str,
+        The name of the dataset representing the movie within the
+        output h5 file.
+    xtable: dict
+        A dict containing aggregate information about the input video
+        to be used in the dewarping process.
+    FOVwidth: int
+        Field of View width. Will be the width of the output image, unless 0,
+        then it will be the width of the input image.
+    movie_shape: tuple[int, int],
+        The shape of the numpy array containing the movie.
+    movie_dtype
+        The data type of the values in the numpy array representing the movie.
+    Returns
+    -------
+    """
+    # Remove old output if it exists and create new output file
+    if os.path.exists(output_file):
+        os.remove(output_file)
+
+    dewarped_file = h5py.File(output_file, 'w')
+    if FOVwidth == 512:
+        dewarped_file.create_dataset(
+            output_dataset, shape=movie_shape, dtype=movie_dtype
+        )
+    else:
+        out_shape = [
```

Comment:
It is not clear to me that this will necessarily satisfy the requirements for using `h5py.create_dataset()`:
https://docs.h5py.org/en/stable/high/dataset.html#creating-datasets

Reply:
Sorry, I missed this one until now. In that documentation, it says "To initialise a dataset, all you have to do is specify a name, shape, and optionally the data type," which is exactly what this line of code does (`output_dataset` is the name of the dataset inside the h5).

Reply:
This is what I was referring to in the docs:
```
 It’s required that (1) the total number of points in shape match the total number of points in data.shape
```
I don't see this function enforcing that the data written has the same shape as specified. Do we even need this argument? I.e. is `output_dataset` being reshaped in this operation? If not, we wouldn't need the `movie_shape` arg at all and we wouldn't need to tell h5py anything special.

Reply:
Oh, I see. Nothing in our code will later check to ensure this restriction, the `h5py` library will simply throw an error if we try to put anything wrong in there.

The data that we save doesn't get reshaped at all (technically it can be a different shape from the input, but at this point in the code we know if that will happen and what the size will be). I just tested it and it looks like it works just fine without specifying the size.

The docs also say 
```An HDF5 dataset created with the default settings will be contiguous; in other words, laid out on disk in traditional C order.```

So I assume that leaving it in would make writing large data sets faster, since it would be able to set aside all the necessary space right at the beginning (knowing the data type and the shape), before doing the dewarping, so it will never have to stop and ask the OS for more space and move the (potentially huge) data around.

I can have this compute the data shape early, then add an assert to the `xdewarp` method before returning the dewarped frame to ensure that it's the right shape. Does that sound good?

Reply:
I see. I misunderstood what this function was doing, nevermind, thanks for following up